pkg/build/builder/common.go

```diff
@@ -182,15 +184,31 @@ func updateBuildRevision(build *buildapi.Build, sourceInfo *git.SourceInfo) *bui
 	}
 }
 
-func retryBuildStatusUpdate(build *buildapi.Build, client client.BuildInterface, sourceRev *buildapi.SourceRevision) error {
-	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+// handleBuildStatusUpdate handles updating the build status
+// retries occur on update conflict and unreachable api server
+func handleBuildStatusUpdate(build *buildapi.Build, client client.BuildInterface, sourceRev *buildapi.SourceRevision) {
+	var latestBuild *buildapi.Build
+	var err error
+
+	updateBackoff := wait.Backoff{
+		Steps:    10,
+		Duration: 25 * time.Millisecond,
+		Factor:   2.0,
+		Jitter:   0.1,
+	}
+
+	updateErr := wait.ExponentialBackoff(updateBackoff, func() (bool, error) {
+		var shouldNotRetry bool
```

Comment:
(the name would make more sense if there were any non-retryable errors, but as it stands, there are not).